### PR TITLE
Support global command hotkeys

### PR DESCRIPTION
**Pull request by Shuhala**
### Support `hotkeys` settings
- Add hotkeys and commands types
- Add two dummy commands with a `console.log` callback in default settings for demo purposes

### Listen to "global" commands 
- Add `useCommands` hook which creates a window event listener for keyboard events
  - When a hotkey is pressed, lookup registered hotkeys to find if we have a match for a command. If yes, trigger the `command.callback`.
  - Note: If two commands share the same hotkey, only the first found will be triggered. We intentionally do not support duplicate hotkeys
- Add `GlobalCommands` component (not sure of the name..). This component is responsible of listening for changes on `settings.hotkeys` and update `useCommands` hook.

### Edit hotkeys
- Add `HotkeysTab` in `SettingsModal` to manage `hotkeys` settings
- Add text field to filter available commands
- Delete a hotkey from a command
- Listen and add a hotkey to a command (no duplicates)

image
image
image

### CODE REVIEW

File: src/ui/hooks/useCommands.ts [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+              ])
+            )
+        )
+      ),
+    [commands]
+  );
+
+  useEffect(() => {
+    const downHandler = ({ key }: KeyboardEvent) => {
+      if (!modifiersPressed.includes(key) && modifierKeys.includes(key)) {
```
Comment (aviau) [inline src/ui/hooks/useCommands.ts:40]: isModifier(key)

**Comment by aviau** [comment]
Can we make is so that we can trigger hotkeys even if we press the modifier at the end of the combo?

File: src/ui/hooks/useCommands.ts [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+ * @param commands
+ */
+export function useCommands(commands: Command[]) {
+  const [modifiersPressed, setModifiersPressed] = useState<Modifier[]>([]);
+  const [keyPressed, setKeyPressed] = useState<string | undefined>();
+  const [activeCommand, setActiveCommand] = useState<Command>();
+
+  const registerCommandHotkeys: Map<string, Command> = useMemo(
+    () =>
+      new Map<string, Command>(
```
Comment (aviau) [inline src/ui/hooks/useCommands.ts:23]: Can we write this in a couple of lines for people like me that can't read?

**Comment by aviau** [comment]
sick

File: src/ui/utilities/hotkeys.ts [hunk 3]
```diff
@@ -0,0 +1,4 @@
+import { Hotkey } from '@/ui/foundation/SettingsProvider';
+
+export const hotkeyToString = (hotkey: Hotkey): string =>
+  `${hotkey.modifiers.join('+')}+${hotkey.key.toUpperCase()}`;
```
Comment (aviau) [inline src/ui/utilities/hotkeys.ts:4]: sorted(hotkey.modifiers) or the same hotkey can have two representations

File: src/ui/hooks/useCommands.ts (hunk 1 above)
Comment (Shuhala) [reply src/ui/hooks/useCommands.ts:40]: Good idea

**Pull request merged**